homework/homework24/tests/test_web_page.py

```diff
@@ -0,0 +1,196 @@
+"""This module tests the creation, updating and deletion of a contact"""
+
+
+import time
+import pytest
+from selenium import webdriver
+from selenium.common import TimeoutException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from homework.homework24.page_locators.login_page \
+    import LoginPageLocators
+from homework.homework24.page_locators.add_contact_page \
+    import AddContactLocators
+from homework.homework24.page_locators.update_contact_details_page \
+    import EditContactLocators
+
+
+URL = 'https://thinking-tester-contact-list.herokuapp.com/'
+EMAIL = 'xesed@mailto.plus'
+PASSWORD = 'qwerty1'
+
+
+@pytest.fixture()
+def open_browser():
+    """
+    Fixture function that opens a browser and navigates to a specified URL
+    """
+    browser = webdriver.Chrome()
+    browser.implicitly_wait(5)
+    browser.get(URL)
+    yield browser
+    browser.quit()
+
+
+@pytest.fixture(autouse=True)
+def login(open_browser):
+    """
+    Logs in to the web page.
+
+    Parameters:
+        open_browser (selenium.webdriver.Chrome):
+        The browser fixture to open the login page.
+    """
+    browser = open_browser
+    wait = WebDriverWait(browser, 10)
+
+    email_field = wait.until(EC.presence_of_element_located(
+        (By.CSS_SELECTOR, LoginPageLocators.input_email)))
+    email_field.send_keys(EMAIL)
+
+    password_field = browser.find_element(
+        By.CSS_SELECTOR, LoginPageLocators.input_password)
+    password_field.send_keys(PASSWORD)
+
+    button_submit = browser.find_element(
+        By.CSS_SELECTOR, LoginPageLocators.submit_btn)
+    button_submit.click()
+
+    wait.until(EC.presence_of_element_located(
+        (By.XPATH, '//*[contains(text(), "Contact List")]')))
+
+
+def test_add_contact(open_browser):
+    """
+    Adds a new contact to the contact list.
+
+    Parameters:
+        open_browser (selenium.webdriver.Chrome):
+        The browser fixture to open the login page.
+    """
+    browser = open_browser
+    wait = WebDriverWait(browser, 10)
+
+    button_add_contact = wait.until(EC.element_to_be_clickable(
+        (By.CSS_SELECTOR, AddContactLocators.add_contact_btn)))
```

Comment:
Define and use locators in a correct way, second part:
```suggestion
    button_add_contact = wait.until(EC.element_to_be_clickable(*AddContactLocators.add_contact_btn))
```

Reply:
removed By.CSS_SELECTOR from calls, since I added it to locators